service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -86,6 +86,24 @@ protected boolean doRegister() {
     if (!StringUtils.isEmpty(serviceId)) {
       // This microservice has been registered, so we just use the serviceId gotten from service center
       microservice.setServiceId(serviceId);
+
+      // Need to update microservice properties if we have modified or added properties of microservices.
+      Microservice microserviceTemp = new Microservice();
+      EnvAdapterManager.INSTANCE.processMicroserviceWithAdapters(microserviceTemp);
+      if (!microserviceTemp.getProperties().isEmpty()) {
+        Map<String, String> propertiesTemp = microserviceTemp.getProperties();
+        if (srClient.getMicroservice(serviceId) != null) {
+          microserviceTemp.setProperties(srClient.getMicroservice(serviceId).getProperties());
+          microserviceTemp.getProperties().putAll(propertiesTemp);
```

Comment:
srClient.getMicroservice(serviceId)  is a remote RPC call, use the result above

Reply:
OK